### PR TITLE
Backport pytest to release-0.23

### PR DESCRIPTION
**Pull request by danielhoherd**
## Description

This change set backports our pytest chart tests to release 0.23. There are a few instances where we have had to make modifications to 0.23, and there are a few more in the pipeline, and having these tests in place will help us gain confidence in changes we make to this LTS release.

Having these tests in place is prerequisite for some customer upgrade work.

This change set is pretty safe. The only customer facing change is tweaking the rbac template. Everything else is just CI tweaks and test additions, and those additions mostly put us in a better testing stance going forward. Also, all of these pytest additions are taken from master branch, so they have already been reviewed.

## Itemized changes

- Backport all relevant pytest chart tests from master branch (some tests were dropped because they were for features that are newer than 0.23)
- Drops k8s 1.16 testing in CI because 1.16 is deprecated in all supported cloud environments (not possible to run 1.16 in any public cloud env)
- Bump ap-build image
- Bump ubuntu machine image and its python version used in testing
- Backport some CI changes from master branch: Makefile, build-helm-chart.sh, pre-commit
- Drop `apiVersion.rbac.v1beta2` template in favor of `apiVersion.rbac` which handles both v1 and v1beta1. There are some hard-coded apiVersion.rbac.v1 strings that should be fixed in newer versions. https://github.com/astronomer/issues/issues/3862

## Related to

- <https://github.com/astronomer/issues/issues/3855>
- <https://github.com/astronomer/issues/issues/3860>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,56 +1,81 @@
...
+	-pre-commit run requirements-txt-fixer --all-files --show-diff-on-failure
+
+.PHONY: show-docker-images
+show-docker-images: ## Show all docker images and versions used in the helm chart
+	@helm template . \
+		--set global.baseDomain=foo.com \
+		--set global.blackboxExporterEnabled=True \
+		--set global.kedaEnabled=True \
+		--set global.postgresqlEnabled=True \
+		--set global.postgresqlEnabled=True \
```
Comment (kaxil) [inline Makefile:76]:
```suggestion
		--set global.postgresqlEnabled=True \
```

Comment (danielhoherd) [reply Makefile:76]: Good eye!

**Review by thmzlt — APPROVED** [review]
Looks straightforward to me.

**Review by kaxil — APPROVED** [review]

**Review by park-peter — APPROVED** [review]
LGTM

**Pull request merged**